app/controllers/api/v1/punches_controller.rb

```diff
@@ -5,11 +5,22 @@ def index
         render json: scoped_punches, status: :ok, each_serializer: PunchSerializer
       end
 
+      def bulk
+        created_punches = BulkPunchesService.call(bulk_punches_params, current_user)
+        render json: created_punches, status: :created, each_serializer: PunchSerializer
+      rescue StandardError => e
```

Comment:
I do not like this type of approach. Avoid using `rescue` to control your data flow.
The `rescue` can generate some unexpected behavior.

Reply:
I didn't think in this way, any error can be sent to the rescue and it should be a problem, thanks.